hack/junit.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# This utility file contains functions that format test output to be parsed into jUnit XML
+
+# os::test::declare_suite_start prints a message declaring the start of a test suite
+function os::test::declare_suite_start() {
+	local suite_name=$1
+	echo "==== BEGIN TEST SUITE ${suite_name} =====" >> "${JUNIT_REPORT_OUTPUT}"
+}
+
+# os::test::declare_suite_end prints a message declaring the end of a test suite
+function os::test::declare_suite_end() {
+	echo "==== END TEST SUITE =====" >> "${JUNIT_REPORT_OUTPUT}"
+}
+
+# os::test::declare_test_start prints a message declaring the start of a test case
+function os::test::declare_test_start() {
+	echo "==== BEGIN TEST CASE =====" >> "${JUNIT_REPORT_OUTPUT}"
+}
+
+# os::test::declare_test_end prints a message declaring the end of a test case
+function os::test::declare_test_end() {
```

Comment:
Keep track of the starts and ends.  I'd like to be able to check depth during `test-cmd.sh` to make sure no one forgot one and to be sure that `test_end` isn't called if we haven't started.  This will help keep us tidy and matching.